### PR TITLE
FormatRBear:  Cleanup R-command 

### PR DESCRIPTION
**Pull request by Asnelchristian**
Closes https://github.com/coala-analyzer/coala-bears/issues/514

### CODE REVIEW

File: bears/r/FormatRBear.py [hunk 1]
```diff
@@ -20,12 +20,12 @@ class FormatRBear:
...
     def create_arguments(filename, file, config_file,
-                         r_keep_comments: bool=True,
+                         r_keep_comments: bool=None,
                          r_keep_blank_lines: bool=True,
                          r_braces_on_next_line: bool=False,
-                         r_use_arrows: bool=False,
+                         r_use_arrows: bool=None,
                          tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
-                         r_max_expression_length: int=20):
+                         r_max_expression_length: int=0):
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: huh wasn't this alread introduced?

File: bears/r/FormatRBear.py [hunk 2]
```diff
@@ -20,12 +20,12 @@ class FormatRBear:
 
     @staticmethod
     def create_arguments(filename, file, config_file,
-                         r_keep_comments: bool=True,
+                         r_keep_comments: bool=None,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: in this case I would keep the default, a certain default config is useful for beginners.
I only wanted you to disable the max-expression-length as it is too advanced to be checked by default imo :)

File: bears/r/FormatRBear.py [hunk 3]
```diff
@@ -20,12 +20,12 @@ class FormatRBear:
...
                          r_keep_blank_lines: bool=True,
                          r_braces_on_next_line: bool=False,
-                         r_use_arrows: bool=False,
+                         r_use_arrows: bool=None,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: same here, keep default :+1:

File: bears/r/FormatRBear.py [hunk 4]
```diff
@@ -68,20 +68,20 @@ def create_arguments(filename, file, config_file,
...
                                           r_braces_on_next_line)))
+        # Disable r_keep_comments if None
+        rcode += ('' if r_keep_comments == None
+                  else ',comment=' + _map_to_r_bool(r_keep_comments))
+        # Disable r_use_arrows if None
+        rcode += ('' if r_use_arrows == None
+                  else ',arrow=' + _map_to_r_bool(r_use_arrows))
         # Disable r_max_expression_length if it is equal to 0
         rcode += ('' if r_max_expression_length == 0
                   else ',width.cutoff=' + str(r_max_expression_length)) + ')'
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: I would go now for a more generic way.
maybe you could do something like this?

```
options = {'source="' + escape(...) + '"': True,
           'comment=' + _map_to_r_bool(r_keep_comments): r_keep_comments is not None,
           'arrow=' + _map_to_r_bool(r_use_arrows): r_use_arrows is not None,
           ...}
rcode = 'library(formatR);formatR::tidy_source({})'.format(
            ','.join(option for option, take_option in options.items() if take_option))
```

then you can easily extend this for all parameters :+1:

File: bears/r/FormatRBear.py (hunk 1 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: it was i guess i messed up with git  little

File: bears/r/FormatRBear.py (hunk 2 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: okay

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: this was already done though

File: bears/r/FormatRBear.py (hunk 3 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: in that case, think there is nothing to be (IMO)
What should i do ?

File: bears/r/FormatRBear.py (hunk 4 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: it is crystal.
but i think there is a problem, having `take_option = r_keep_comments`(assuming r_keep_comments is not None)  means that `comment=TRUE` is way `comment` can be passed to `tidy_source`

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: I kept the `None` because, IMO there is no need passing an option to tidy_source unless it is different from the default.
If you agree with this (i will do a simple modification on what i just pushed)

File: bears/r/FormatRBear.py [hunk 5]
```diff
@@ -65,24 +65,20 @@ def create_arguments(filename, file, config_file,
...
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"': True,
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines): True,
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line):
+                   True,
+                   'comment=' + _map_to_r_bool(r_keep_comments):
+                   True if r_keep_comments is not None else False,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: please don't do that -->
`r_keep_comments is not None` evaluates to True or False :+1:

File: bears/r/FormatRBear.py (hunk 4 above)
Comment (Makman2) [inline bears/r/FormatRBear.py]: I don't understand what you mean :O

File: bears/r/FormatRBear.py [hunk 6]
```diff
@@ -65,24 +65,20 @@ def create_arguments(filename, file, config_file,
...
+                   'arrow=' + _map_to_r_bool(r_use_arrows):
+                   True if r_use_arrows is not None else False,
+                   'width.cutoff=' + str(r_max_expression_length):
+                   True if r_max_expression_length is not 0 else False,
+                   'indent=' + str(tab_width): True}
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: please add indent to the values by one level:
```
'the_option=' + option_val:
    True/False/whatever
```

File: bears/r/FormatRBear.py [hunk 7]
```diff
@@ -65,24 +65,20 @@ def create_arguments(filename, file, config_file,
...
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"': True,
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines): True,
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line):
+                   True,
+                   'comment=' + _map_to_r_bool(r_keep_comments):
+                   True if r_keep_comments is not None else False,
+                   'arrow=' + _map_to_r_bool(r_use_arrows):
+                   True if r_use_arrows is not None else False,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: same here

File: bears/r/FormatRBear.py [hunk 8]
```diff
@@ -65,24 +65,20 @@ def create_arguments(filename, file, config_file,
...
+                   'width.cutoff=' + str(r_max_expression_length):
+                   True if r_max_expression_length is not 0 else False,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: same here

File: bears/r/FormatRBear.py [hunk 9]
```diff
@@ -65,24 +65,20 @@ def create_arguments(filename, file, config_file,
...
-                                          r_keep_blank_lines),
-                                      r_use_arrows=_map_to_r_bool(
-                                          r_use_arrows),
-                                      r_braces_on_next_line=_map_to_r_bool(
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"': True,
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines): True,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: so you want to explicitly pass that everytime to `tidy_source`?

File: bears/r/FormatRBear.py [hunk 10]
```diff
@@ -65,24 +65,20 @@ def create_arguments(filename, file, config_file,
...
+                   True if r_keep_comments is not None else False,
+                   'arrow=' + _map_to_r_bool(r_use_arrows):
+                   True if r_use_arrows is not None else False,
+                   'width.cutoff=' + str(r_max_expression_length):
+                   True if r_max_expression_length is not 0 else False,
+                   'indent=' + str(tab_width): True}
+
+        rcode = 'library(formatR);formatR::tidy_source({})'.format(
+            ','.join(option for option, take_option in
+                     options.items() if take_option))
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: I like this style:
```
','.join(option
         for option, take_option in options.items()
         if take_option)
```
but this is just preference^^

File: bears/r/FormatRBear.py (hunk 2 above)
Comment (Makman2) [inline bears/r/FormatRBear.py]: you haven't changed this and the comments below :O
so just remove that change, having `r_keep_comments=True` is totally resonable^^

File: bears/r/FormatRBear.py (hunk 9 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: yes

File: bears/r/FormatRBear.py [hunk 11]
```diff
@@ -65,24 +65,21 @@ def create_arguments(filename, file, config_file,
...
-                                      r_use_arrows=_map_to_r_bool(
-                                          r_use_arrows),
-                                      r_braces_on_next_line=_map_to_r_bool(
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"': True,
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines):
+                       r_keep_blank_lines is not None,
```
Comment (sils1297) [inline bears/r/FormatRBear.py]: they all can't be None since they will be a bool right? So this always evaluates to false

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: isn't it the reverse,

2016-06-16 20:58 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

Comment (sils1297) [inline bears/r/FormatRBear.py]: yes, this\ll always be true, right, still makes the expression useless, right?

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: the only possiblility for it to be false is to have `r_keep_blank_lines ==
None` that is why i am thinking of removing that, and setting `None` as
default for `r_keep_comments` and `r_use_arrows` :)

2016-06-16 21:18 GMT+01:00 Christian Ngoulla Sob <ngoullasob@gmail.com>:

Comment (sils1297) [inline bears/r/FormatRBear.py]: but the user can't set it to None because you're asking coala to convert it to a bool for you right? And you don't give a None as default

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: i meant having something like `r_use_comments: bool=None` in the function
declaration

2016-06-16 21:30 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

Comment (sils1297) [inline bears/r/FormatRBear.py]: That would be better

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: Done :)

2016-06-16 21:51 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

File: bears/r/FormatRBear.py (hunk 2 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: `tidy_source` already sets `r_keep_comments` to `true`.
If i want to effectively use the `dict` you gave below, then  `take_option` is false only when 
`r_keep_comments: bool=None` in the function declaration.

Comment (Makman2) [inline bears/r/FormatRBear.py]: you think we should make everything `None` and use the default style of `formatR`? I fear that the default style of formatR might change depending on version^^

File: bears/r/FormatRBear.py (hunk 1 above)
Comment (Makman2) [inline bears/r/FormatRBear.py]: hm seems git is right. Wanna do another commit that sets the default?

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: actually theres is no default value for this one, it must be passed to
`tidy_source` explicitly and its value varies from 20 to 500 :) should i
mention this somewhere? :)

2016-06-18 19:49 GMT+01:00 Mischa Krüger <notifications@github.com>:

File: bears/r/FormatRBear.py (hunk 2 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: humm... i didn't think about that?

2016-06-18 19:47 GMT+01:00 Mischa Krüger <notifications@github.com>:

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: What should i do....

File: bears/r/FormatRBear.py [hunk 12]
```diff
@@ -68,24 +68,15 @@ def create_arguments(filename, file, config_file,
...
-                                          r_keep_blank_lines),
-                                      r_use_arrows=_map_to_r_bool(
-                                          r_use_arrows),
-                                      r_braces_on_next_line=_map_to_r_bool(
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"',
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
```
Comment (AbdealiJK) [inline bears/r/FormatRBear.py]: Just `r_bool` is probably a simpler name compared to `_map_to_r_bool` ... although it's not in the scope of this PR.
Could you make an issue for it ?

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: okay :)

2016-06-21 12:23 UTC+01:00, AbdealiJK <notifications@github.com>:

**Comment by Adrianzatreanu** [comment]
ack eaa0d17588

File: bears/r/FormatRBear.py [hunk 13]
```diff
@@ -68,24 +68,15 @@ def create_arguments(filename, file, config_file,
...
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
+        options |= ({'width.cutoff=' + str(r_max_expression_length)}
+                    if r_max_expression_length != 0 else set())
+
+        rcode = 'library(formatR);formatR::tidy_source({})'.format(
+            ','.join(option for option in options))
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: -->
```
','.join(options)
```

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: ah okay :)

2016-06-22 20:10 GMT+01:00 Mischa Krüger <notifications@github.com>:

File: bears/r/FormatRBear.py [hunk 14]
```diff
@@ -68,24 +68,15 @@ def create_arguments(filename, file, config_file,
...
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"',
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
+        options |= ({'width.cutoff=' + str(r_max_expression_length)}
+                    if r_max_expression_length != 0 else set())
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: in this case I would do
```
if r_max_expression_length != 0:
    options.add('width.cutoff=' + str(r_max_expression_length))
```

File: bears/r/FormatRBear.py [hunk 15]
```diff
@@ -68,24 +68,15 @@ def create_arguments(filename, file, config_file,
...
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"',
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: but now these can't be disabled completely right?
This commit is good anyway :+1: , though it requires a commit message change

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: any suggestion for the commit message? :)


2016-06-22 20:30 GMT+01:00 Mischa Krüger <notifications@github.com>:

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/525/commits/dbde4c23525d74f2eb97fb8f1ddf9a84aef78b5e

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/525/commits/91ea80fc833fd4b0996fb5b740abe72ce1a0ce07

File: bears/r/FormatRBear.py [hunk 16]
```diff
@@ -29,7 +29,7 @@ def create_arguments(filename, file, config_file,
                          r_braces_on_next_line: bool=False,
                          r_use_arrows: bool=False,
                          tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
-                         r_max_expression_length: int=20):
+                         r_max_expression_length: int=0):
```
Comment (hypothesist) [inline bears/r/FormatRBear.py]: Commit message should read ```... value of ``r_max_expression_length`` to
zero, ...```, not `r_max_expression`

**Comment by hypothesist** [comment]
https://github.com/coala-analyzer/coala-bears/pull/525/commits/4d32ad26c4619accd2e0e366e70d15a33843296d needs work

**Comment by Adrianzatreanu** [comment]
unack 20e7e3f64

File: bears/r/FormatRBear.py [hunk 17]
```diff
@@ -69,24 +69,15 @@ def create_arguments(filename, file, config_file,
...
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"',
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
+        if r_max_expression_length != 0:
```
Comment (Adrianzatreanu) [inline bears/r/FormatRBear.py]: cant you leave the ``!= 0`` out? thats what if does by default, checks if the expresion is ``!=0``

**Comment by Adrianzatreanu** [comment]
did you change the commit message of the first commit, the cleanup one already? cuz its not really helpful or meaningful to me, I might be wrong though and it might be my fault :) but i think a more suggestive one would be great

**Comment by Asnelchristian** [comment]
this is not done by default by ``tidy_source`` we are the one trying to
disable it by setting it to 0.

2016-06-27 16:37 GMT+01:00 Zatreanu Adrian-Gabriel <notifications@github.com

File: bears/r/FormatRBear.py (hunk 17 above)
Comment (Adrianzatreanu) [inline bears/r/FormatRBear.py]: bumping as you haven't answered this

**Comment by Adrianzatreanu** [comment]
ack 0117ff4b8

File: bears/r/FormatRBear.py (hunk 16 above)
Comment (Adrianzatreanu) [inline bears/r/FormatRBear.py]: bumping this as it wasnt done either

**Comment by Adrianzatreanu** [comment]
hey you still havent changed/answered 2 things that were mentioned in comments :) not acking until then

File: bears/r/FormatRBear.py (hunk 17 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: this is not done by default by ``tidy_source``, @Makman2  proposed that we should disable it by 
default, and we do so by if its value is ``0``

Comment (Makman2) [inline bears/r/FormatRBear.py]: @Adrianzatreanu means to change
```python
if r_max_expression_length != 0:
```
to
```python
if r_max_expression_length:
```
as `bool(some_int)` is effectively this:
```
def __bool__(some_int):
    return some_int != 0
```
:)

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: ah okay :)
Done :)

2016-06-28 11:50 GMT+01:00 Mischa Krüger <notifications@github.com>:

File: bears/r/FormatRBear.py (hunk 15 above)
Comment (Makman2) [inline bears/r/FormatRBear.py]: oh damn missed this comment, sorry.
Maybe `FormatRBear: Use set for `tidy_source` parameters`?

**Comment by Makman2** [comment]
ack 2866bd3

File: bears/r/FormatRBear.py (hunk 16 above)
Comment (Makman2) [inline bears/r/FormatRBear.py]: I think this suffices:
```
FormatRBear: Disable 'r_max_expression_length'
```
btw: for the commit body you are allowed to use 73 chars per line (or 72, ahh just use 72 :3). Makes stuff better readable :+1:

Comment (Makman2) [inline bears/r/FormatRBear.py]: and you may want to include the reason inside the commit message body :+1: :)

File: bears/r/FormatRBear.py [hunk 18]
```diff
@@ -69,24 +69,15 @@ def create_arguments(filename, file, config_file,
...
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
+        if r_max_expression_length:
+            options.add('width.cutoff=' + str(r_max_expression_length))
+
+        rcode = 'library(formatR);formatR::tidy_source({})'.format(
+            ','.join(options))
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: Commit message shortlog still not changed + `Closes xxx` is not appropriate any more :+1:

**Comment by Makman2** [comment]
unack c3d8ea9

**Comment by Makman2** [comment]
ack 98aeff1 4af9c40

**Comment by Makman2** [comment]
ack 123f415 b29539e 89627a9

**Comment by Makman2** [comment]
@rultor merge
